projects/Mallard/src/authentication/authorizers/__tests__/identity-authorizer.spec.ts

```diff
@@ -1,5 +1,9 @@
 import { getUserName, detectAuthType } from '../IdentityAuthorizer'
 
+jest.mock('src/helpers/locale', () => ({
+    locale: () => jest.fn().mockReturnValue('en_GB'),
+}))
```

Comment:
I see this mock has added in many test classes and no other changes in those classes (i.e. no new test). If you can enlighten my js testing knowledge by explaining the reason of adding these mocks that would be great :)

Reply:
In this case, that particular function requires access to the device's OS to get the locale. The consuming function in the test doesn't care about this, just the result. This keeps the test focussed and reduces the possibility of external influence.

Reply:
Frustrating thing here is this would be best to keep at the top level as a mock in the `__mocks__` folder but it doesn't seem to register because its not an external module

Reply:
It there something in the jest config file we can play with to make it work in `__mocks__`?

Reply:
Thanks @mohammad-haque. I have just done a test of this approach and locale returns `undefined` rather than `en_GB`. This is because at the test file level, you are essentially creating a blank mock rather than a mock that returns a value. It does the same job but makes it less predictable. Im going to revert this.

Reply:
That is strange that it doesn't behave the way it described in the documentation! My understanding from the documentation was that at test file level it just activates the mock that has been declared in the __mocks__ folder. But as you said it doesn't do that.